adapter/src/main/java/com/twilio/kudu/sql/rules/KuduNestedJoinRule.java

```diff
@@ -81,24 +87,32 @@ public Boolean visitInputRef(final RexInputRef inputRef) {
       }
     };
 
-    final Boolean isValid = condition.accept(validateJoinCondition);
-
-    // @TODO: Hack for the moment.
-    final boolean containsActorSid = join.getRight().getRowType().getFieldList().stream()
-        .anyMatch(fl -> fl.getName().equalsIgnoreCase("actor_sid"));
-
-    return isValid != null && isValid && containsActorSid;
+    return condition.accept(validateJoinCondition);
```

Comment:
`validateJoinCondition` can return `null`. I believe you need to check for null

Reply:
done